code/street-banner/street-banner.js

```diff
@@ -406,170 +406,161 @@ $(document).on("knack-scene-render.scene_1234", function () {
     location.reload();
   })
 });
-
-
 // lamppost application details
-$(document).on('knack-view-render.view_3664', function(event, page) {
-  $('#view_3664').hide()
-  // Set current record ID to fetch invoices from Knack API
-  var hrefArray = window.location.href.split("/");
-  var recordId = hrefArray[hrefArray.length - 2];
-
-  var payload={
-    "allowed_payment_methods": [
-      "CARD",
-      "BANK"
-    ],
-    "cancel_url": {
-      "url": window.location.href,
-      "label": "Cancel"
+$(document).on("knack-view-render.view_3664", function (event, page, view) {
+  $("#view_3664").hide();
+  // operating under the expectation that there is only one line item to add
```

Comment:
I'm surely over-focused on the word "expectation" here, but maybe it'd be good to ensure that there is only one item by inspecting `view`'s length and `throw`ing an error if it's `> 1` here? Maybe it's a `try / catch` type thing -- i'm not sure how JQuery interacts with these error mechanisms.

If this isn't a big deal - pls ignore of course!

Reply:
I am kind of hesitant to throw any error here, I am going to check if there is way to limit the items before it gets to this point, with Knack rules for example.